internal/pkg/levee/testdata/src/example.com/tests/colocation/tests.go

```diff
@@ -16,22 +16,118 @@ package colocation
+	colocateStringPointer(s, strptr)
+	core.Sink(strptr) // want "a source has reached a sink"
+}
+
+func TestPointerToBasicTypeIsTainted(s core.Source, str string) {
+	// This test is failing because &x introduces an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocateStringPointer(s, &str)
+	core.Sink(str) // TODO want "a source has reached a sink"
+}
+
+func colocateInnoc(core.Source, core.Innocuous) {}
+
+func TestNamedStructTypeIsNotTainted(s core.Source, i core.Innocuous) {
+	colocateInnoc(s, i)
+	core.Sink(i)
+}
+
+func colocateInnocPtr(core.Source, *core.Innocuous) {}
+
+func TestNamedStructPointerIsTainted(s core.Source, i *core.Innocuous) {
+	colocateInnocPtr(s, i)
+	core.Sink(i) // want "a source has reached a sink"
+}
+
+func TestPointerToNamedStructIsTainted(s core.Source, i core.Innocuous) {
+	// This test is failing because &x introduces an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocateInnocPtr(s, &i)
+	core.Sink(i) // TODO want "a source has reached a sink"
+}
+
+type PointerHolder struct{ ptr *core.Source }
+
+func colocatePointerHolder(core.Source, PointerHolder) {}
+
+func TestNamedStructPointerHolderIsTainted(s core.Source, ph PointerHolder) {
+	// This test is failing because ph is created by an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocatePointerHolder(s, ph)
+	core.Sink(ph) // TODO want "a source has reached a sink"
+}
+
+func colocateArrOfValues(core.Source, [1]string) {}
+
+func TestArrOfValuesIsNotTainted(s core.Source, arr [1]string) {
+	colocateArrOfValues(s, arr)
+	core.Sink(arr)
+}
+
+func colocateArrOfPointers(core.Source, [1]*string) {}
+
+func TestArrOfPointersIsTainted(s core.Source, arr [1]*string) {
+	colocateArrOfPointers(s, arr)
+	core.Sink(arr) // want "a source has reached a sink"
+}
+
+func colocateReferenceCollections(core.Source, map[string]string, chan string, []string) {}
+
+func TestReferenceCollectionsAreTainted(s core.Source) {
+	m := make(map[string]string)
+	c := make(chan string)
+	sl := make([]string, 0)
+	colocateReferenceCollections(s, m, c, sl)
+	core.Sink(m)  // want "a source has reached a sink"
+	core.Sink(c)  // want "a source has reached a sink"
+	core.Sink(sl) // want "a source has reached a sink"
+}
+
+func colocateReflectValue(core.Source, reflect.Value) {}
+
+func TestReflectValuesAreTainted(s core.Source, r reflect.Value) {
+	// This test is failing because r is created by an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocateReflectValue(s, r)
+	core.Sink(r) // TODO want "a source has reached a sink"
 }
 
-func TestTaintIsPropagatedToColocatedPointerArgumentsThroughEface(s core.Source, i core.Innocuous, ip *core.Innocuous) {
-	taintColocatedEface(s, &i, ip)
-	core.Sink(s)  // want "a source has reached a sink"
-	core.Sink(i)  // TODO want "a source has reached a sink"
-	core.Sink(ip) // want "a source has reached a sink"
+func colocateSingleEface(s core.Source, e interface{})             {}
```

Comment:
Does `Eface` have a conventional?  Searching around for it, I find it used as the `runtime` unexported type to hold interfaces, which fits here.  I just wanted to be on the same page, and might consider using `Interface` to avoid any confusion at the low cost of four characters.